### PR TITLE
Modify the forward rendering process and remove UIFlow.

### PR DESCRIPTION
**Pull request by caryliu1999**
cocos2dx-lite：https://github.com/cocos-creator/cocos2d-x-lite/pull/3117
editor: https://github.com/cocos-creator/editor-3d/pull/4854

Changes:
1. Remove RenderView and UIBatchModel.
2. Remove UIFlow and UIStage.
3. Add UI phase to forward stage.
4. Split camera UBO from global UBO.
5. Modify the traversal process of the camera. 

fix issues:
https://github.com/cocos-creator/3d-tasks/issues/5083
https://github.com/cocos-creator/3d-tasks/issues/5009

### CODE REVIEW

File: cocos/core/pipeline/index.jsb.ts [hunk 1]
```diff
@@ -27,20 +27,33 @@
...
 export const PipelineStateManager = nr.PipelineStateManager;
+
+const RenderWindowProto = RenderWindow.prototype;
+Object.assign(RenderWindowProto, {
+    getRenderCameras() {
+        const cameras = [];
+        for (let j = 0; j < this._cameras.length; j++) {
+            const camera = this._cameras[j];
+            if (camera.enabled) {
+                camera.update();
```
Comment (minggo) [inline cocos/core/pipeline/index.jsb.ts:53]: It is strange to do camera.update() in get function.

File: cocos/core/pipeline/render-pipeline.ts [hunk 2]
```diff
@@ -91,7 +90,7 @@ export abstract class RenderPipeline extends Asset {
      */
     @displayOrder(0)
     @serializable
-    protected _tag: number = 0;
```
Comment (minggo) [inline cocos/core/pipeline/render-pipeline.ts:94]: Why remove type information?

Comment (caryliu1999) [reply cocos/core/pipeline/render-pipeline.ts:94]: > Why remove type information?

It's auto removed by ESLint save. "Type string trivially inferred from a string literal, remove type annotation."

**Review by LinYunMo — APPROVED** [review]
LGTM

**Review by YunHsiao — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Comment by minggo** [comment]
目前该 PR 会导致 metal 崩溃，不能合并。

**Comment by caryliu1999** [comment]
> 目前该 PR 会导致 metal 崩溃，不能合并。

metal 的问题已修复提交

**Pull request merged**